### PR TITLE
client: Add DialEarlyContext and DialAddrEarlyContext API

### LINKED ISSUE #2812: Allow passing in context when using 0RTT
Hi there,

My team is trying to utilize the 0RTT feature from QUIC.  We use context cancellation in our code. However, I have noticed that `quic-go` provides `DialAddrContext` but no `DialAddrEarlyContext`. Is it possible that we can add another method for `DialAddrEarly` that can take in a context?

### PR DESCRIPTION
**Pull request by VinozzZ**
Fixes #2812.

### CODE REVIEW

**Review by marten-seemann — CHANGES_REQUESTED** [review]
Thank you! This lgtm, just one suggestion:

Can you use `DialAddrEarlyContext` in `DialAddrEarly`, in order to avoid code duplication?
```go
func DialAddrEarly(...) {
     return DialAddrEarlyContext(context.Background(), ....)
}

File: client.go [hunk 1]
```diff
@@ -68,7 +68,23 @@ func DialAddrEarly(
 	tlsConf *tls.Config,
 	config *Config,
 ) (EarlySession, error) {
-	sess, err := dialAddrContext(context.Background(), addr, tlsConf, config, true)
+	sess, err := DialAddrEarlyContext(context.Background(), addr, tlsConf, config)
+	if err != nil {
+		return nil, err
+	}
+	utils.Logger.WithPrefix(utils.DefaultLogger, "client").Debugf("Returning early session")
+	return sess, nil
```
Comment (marten-seemann) [inline client.go:76]: You already have that log statement in `DialAddrEarlyContext`. `DialAddrEarly` can be as simple as just a `return DialAddrEarlyContext(context.Background(), ...)`.

**Review by marten-seemann — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (VinozzZ) [reply client.go:76]: My bad. Just fixed it

File: client.go [hunk 2]
```diff
@@ -68,7 +68,23 @@ func DialAddrEarly(
 	tlsConf *tls.Config,
 	config *Config,
 ) (EarlySession, error) {
-	sess, err := dialAddrContext(context.Background(), addr, tlsConf, config, true)
+	sess, err := DialAddrEarlyContext(context.Background(), addr, tlsConf, config)
```
Comment (marten-seemann) [inline client.go:71]:
```go
return DialAddrEarlyContext(context.Background(), addr, tlsConf, config)
```

**Review by marten-seemann — CHANGES_REQUESTED** [review]

**Review by marten-seemann — APPROVED** [review]
Thank you!

**Pull request merged**